systems/plants/@Manipulator/Manipulator.m

```diff
@@ -36,6 +36,73 @@
       C_times_v = C-G;
     end
     
+    function [xdot,dxdot] = Hdynamics(obj,t,x,u)
+    % Provides the DrakeSystem interface to the manipulatorDynamics.
+    %the dynamics left multiplied by H
+
+      q = x(1:obj.num_positions);
+      v = x(obj.num_positions+1:end);
+
+      if (nargout>1)
+        if ~isempty(obj.position_constraints) || ~isempty(obj.velocity_constraints)
+          % by naming this 'MATLAB:TooManyOutputs', geval will catch the
+          % error and use TaylorVarInstead
+          error('MATLAB:TooManyOutputs','User gradients for constrained dynamics not implemented yet.');
+        end
+
+        % Note: the next line assumes that user gradients are implemented.
+        % If it fails, then it will raise the same exception that I would
+        % want to raise for this method, stating that not all outputs were
+        % assigned.  (since I can't write dxdot anymore)
+        [H,C,B,dH,dC,dB] = obj.manipulatorDynamics(q,v);
+        
+        %ANDY CHANGE
+        %Hinv = inv(H);
+        %Hinv = invmsspoly(H);
+
+        if (obj.num_u>0)
+          vdot = (B*u-C);
```

Comment:
OK, I understand now. I'm not sure if it makes sense to have a new method in Manipulator for this, as you can already get what you need in two lines when you need it:
```
[~, C, B] = obj.manipulatorDynamics(q, v);
mass_matrix_times_vdot = B * u - C;
```
Getting rid of this method would avoid having to find a better name for it, and to me it seems the use of the method is limited to this msspoly input case, which you can specialize on all you want in your own code.